v2/tools/generator/internal/codegen/pipeline/create_backwards_compatibility_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package pipeline
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel"
+)
+
+const CreateBackwardCompatibilityTypesStageID = "createStorageTypes"
+
+// CreateBackwardCompatibilityTypes returns a pipeline stage that creates copies of storage types in other packages to
+// provide backward compatibility with previous releases of Azure Service Operator.
+// Backward compatibility versions are created for all storage versions to allow users of older versions of the operator
+// to easily upgrade. We lean on the fact that storage versions have already been modified to increase compatibility -
+// everything is optional, all enums have been rendered into their underlying primitive types, and so on.
+func CreateBackwardCompatibilityTypes(prefix string) *Stage {
```

Comment:
This is only creating the storage types. Do we also need to create the non-storage types too? If we want to allow people to continue submitting v1alpha1 types I think we would, although maybe we don't need that...

There's also the reality that they have the CRDs in their cluster already (since they installed them previously), and AFAIK they won't get removed by anything, so it might be ok for us to omit them?

If they did provision a new cluster and install `beta0` on it, and then attempt to port over their `v1alpha1` YAML, it would be rejected without the corresponding API types, right? Is that the desired behavior or no? (I can see arguments for both ways).

Reply:
Discussed offline and spec changed; we'll keep the API types for the alpha versions to avoid breaking existing devops scenarios.